### PR TITLE
Spotinst - add emr module

### PR DESCRIPTION
**Pull request by jeffnoehren**
##### SUMMARY
Adding new Spotinst module - EMR

##### ISSUE TYPE
- New Module Pull Request

##### ADDITIONAL INFORMATION
```paste below
ok: [localhost] => {
    "result": {
        "changed": true, 
        "failed": false, 
        "group_id": "smrs-35124875", 
        "message": "Created Ocean Cluster successfully."
    }
}
```

### CODE REVIEW

File: test/sanity/validate-modules/ignore.txt [hunk 1]
```diff
@@ -324,6 +324,7 @@ lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py E322
 lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py E323
 lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py E324
 lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py E326
+lib/ansible/modules/cloud/spotinst/spotinst_mrScaler.py E326
```
Comment (mattclay) [inline test/sanity/validate-modules/ignore.txt]: New ignores should not be added. Correct the underlying issue in the module. If the issue is in an existing module_utils, fix that issue in a separate PR first.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/spotinst/spotinst_mrScaler.py [hunk 2]
```diff
@@ -0,0 +1,933 @@
+#!/usr/bin/python
```
Comment (mattclay) [inline lib/ansible/modules/cloud/spotinst/spotinst_mrScaler.py]: Use lowercase letters for module names.

**Review by mattclay — DISMISSED** [review]

File: test/sanity/validate-modules/ignore.txt (hunk 1 above)
Comment (jeffnoehren) [reply test/sanity/validate-modules/ignore.txt]: removed!

File: lib/ansible/modules/cloud/spotinst/spotinst_mrScaler.py (hunk 2 above)
Comment (jeffnoehren) [reply lib/ansible/modules/cloud/spotinst/spotinst_mrScaler.py]: module name has been changed! thanks for letting me know

**Comment by talzur** [comment]
shipit

**Comment by kmcgrath** [comment]
@mattclay Is there anything else you need from our side to get this task in the release flow?

File: changelogs/fragments/spotinst-add-emr-resource.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - spotinst - Added EMR resource
```
Comment (mattclay) [inline changelogs/fragments/spotinst-add-emr-resource.yml:2]: New modules should not have changelog entries.

**Pull request closed without merging**